public/index.ts

```diff
@@ -976,40 +980,49 @@ function goToZSlice(slice: number): boolean {
   // update UI if successful
 }
 
-function createTestVolume() {
-  const imgData: ImageInfo = {
-    name: "AICS-10_5_5",
+function concatenateArrays(arrays: Uint8Array[]): Uint8Array {
+  const totalLength = arrays.reduce((acc, arr) => acc + arr.length, 0);
+  const result = new Uint8Array(totalLength);
+  let offset = 0;
+  for (const arr of arrays) {
+    result.set(arr, offset);
+    offset += arr.length;
+  }
+  return result;
+}
 
-    originalSize: new Vector3(64, 64, 64),
-    atlasTileDims: new Vector2(8, 8),
-    volumeSize: new Vector3(64, 64, 64),
-    subregionSize: new Vector3(64, 64, 64),
-    subregionOffset: new Vector3(0, 0, 0),
-    physicalPixelSize: new Vector3(1, 1, 1),
+function createTestVolume(): RawArrayLoaderOptions {
+  const sizeX = 64;
+  const sizeY = 64;
+  const sizeZ = 64;
+  const imgData: RawArrayInfo = {
+    name: "AICS-10_5_5",
+    sizeX,
+    sizeY,
+    sizeZ,
+    sizeC: 3,
+    physicalPixelSize: [1, 1, 1],
     spatialUnit: "",
-
-    numChannels: 3,
     channelNames: ["DRAQ5", "EGFP", "SEG_Memb"],
-
-    times: 1,
-    timeScale: 1,
-    timeUnit: "",
-
-    numMultiscaleLevels: 1,
-    multiscaleLevel: 0,
-
-    transform: { translation: new Vector3(0, 0, 0), rotation: new Vector3(0, 0, 0) },
   };
 
   // generate some raw volume data
   const channelVolumes = [
-    VolumeMaker.createSphere(imgData.subregionSize.x, imgData.subregionSize.y, imgData.subregionSize.z, 24),
-    VolumeMaker.createTorus(imgData.subregionSize.x, imgData.subregionSize.y, imgData.subregionSize.z, 24, 8),
-    VolumeMaker.createCone(imgData.subregionSize.x, imgData.subregionSize.y, imgData.subregionSize.z, 24, 24),
+    VolumeMaker.createSphere(sizeX, sizeY, sizeZ, 24),
+    VolumeMaker.createTorus(sizeX, sizeY, sizeZ, 24, 8),
+    VolumeMaker.createCone(sizeX, sizeY, sizeZ, 24, 24),
   ];
+  const alldata = concatenateArrays(channelVolumes);
   return {
-    imgData: imgData,
-    volumeData: channelVolumes,
+    metadata: imgData,
+    data: {
+      // expected to be "uint8" always
+      dtype: "uint8",
+      // [c,z,y,x]
+      shape: [channelVolumes.length, sizeZ, sizeY, sizeX],
+      // the bits (assumed uint8!!)
+      buffer: new DataView(alldata.buffer),
+    } as RawArrayData,
```

Comment:
Do you need the type coercion here for `RawArrayData`? I would expect Typescript to infer the type because of the function's return type; if `RawArrayData` ever changes, the mismatch would not get highlighted here.